From 4a0c142ef8d3569194da2dcc5a60cfa968523cbb Mon Sep 17 00:00:00 2001
From: Sima Baymani <sima.baymani@gmail.com>
Date: Sun, 16 Mar 2014 22:16:19 +0100
Subject: [PATCH 4/4] Close stdin before killing cat-process
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

On xfce4-terminal, the call to kill the git cat-file process is not sufficient:
the process does not exit and the caller is not returned to the prompt.

Adding a call to close stdin makes the process exit cleanly.

Suggested-by: Christian Häggström <kalvdans@gmail.com>
Signed-off-by: Sima Baymani <sima.baymani@gmail.com>
Signed-off-by: Catalin Marinas <catalin.marinas@gmail.com>
---
 stgit/lib/git.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/stgit/lib/git.py b/stgit/lib/git.py
index 37a1989..8addc46 100644
--- a/stgit/lib/git.py
+++ b/stgit/lib/git.py
@@ -578,6 +578,7 @@ class CatFileProcess(object):
     def __shutdown(self):
         p = self.__proc
         if p:
+            p.stdin.close()
             os.kill(p.pid(), signal.SIGTERM)
             p.wait()
     def cat_file(self, sha1):
-- 
1.9.1

